### PR TITLE
fuzzy search with slop

### PR DESCRIPTION
**Pull request by yanaiela**
Added two functions that allow to use fuzzy search with ES.
They do so by using the `slop` argument, that allows to use match strings with some (fixed) distance in between them.
Wrote a count and a get-document functions

### CODE REVIEW

File: wimbd/es/__init__.py [hunk 1]
```diff
@@ -406,3 +407,160 @@ def count_total_occurrences_of_unigrams(
...
+    all_phrases: bool = False,
+    slop: int = 5,
+    num_documents: int = 10,
+    return_all_hits: bool = False,
+    sort_field: str = "date",
+    subset_filter: Optional[Dict[str, Any]] = None,
+    es: Optional[Elasticsearch] = None,
+) -> Generator[Dict, None, None]:
+    """
+
```
Comment (AkshitaB) [inline wimbd/es/__init__.py]: Missing docstring; helpful to add what `slop` does and how to set the value.

File: wimbd/es/__init__.py [hunk 2]
```diff
@@ -406,3 +407,160 @@ def count_total_occurrences_of_unigrams(
         )
         term_freq_dict[term] = total_freq
     return term_freq_dict
+
+
+def _query_documents_contain_phrase_with_slop(phrases: str,
```
Comment (AkshitaB) [inline wimbd/es/__init__.py]: Is it possible to merge this function with `_query_documents_contain_phrase` ? The default value for `slop` would be 0, I think?

File: wimbd/es/__init__.py (hunk 1 above)
Comment (AkshitaB) [reply wimbd/es/__init__.py]: Similar to `_query...` , can this be merged with `get_documents_containing_phrases`?

File: wimbd/es/__init__.py [hunk 3]
```diff
@@ -406,3 +407,160 @@ def count_total_occurrences_of_unigrams(
...
+    es: Optional[Elasticsearch] = None,
+) -> Dict[str, int]:
+    """
+    Count the number of documents matching each string with a specified slop.
+
+    :param index: Name of the index
+    :param phrases: A single string or a list of strings to be matched in the `text` field.
+    :param all_phrases: Whether the document should contain all phrases (AND clause) or any
+        of the phrases (OR clause).
+    :param slop: The number of words allowed between the tokens in the string.
```
Comment (AkshitaB) [inline wimbd/es/__init__.py]: So in this case, we're looking for documents where the phrases "legal based" and "licenses" occur within 3 words of each other?

**Review by AkshitaB — CHANGES_REQUESTED** [review]

File: wimbd/es/__init__.py (hunk 1 above)
Comment (yanaiela) [reply wimbd/es/__init__.py]: oh yeah good point. I think it should be possible

**Review by AkshitaB — APPROVED** [review]
LGTM!

**Pull request merged**